### ISSUE TITLE
New URLs within the same domain are also blocked

### ISSUE THREAD

**Issue by arthurcgusmao** [Ishmaeel/CrackbookRevival#1]
First of all, thanks for the good work of reviving this very useful chrome extension.

I am having the following issue: whenever I open a new tab from a junk site (e.g. http://facebook.com) or when I access another URL within the same domain (e.g. http://facebook.com/myprofile) the delay is applied once again.

I don't know if it is intended, but what I think would be the best solution in this case is the delay should not be applied as long as we have at least one open tab of the respective junk site.

Then, if at some point all tabs of the junk site were closed, the next time we open one the delay should be applied again.

Does this make sense?
Thanks for your attention and support.

-----
Edit: Another extension named *Delayed Gratification* has a functionality that addresses this problem of blocking new tabs: it is possible to set up a timer that only blocks new open junk sites after a configured interval. This let's you access the junk site and keep opening new tabs freely for a while.

**Comment by Ishmaeel** [comment]
Thanks for the report. Yes, as I understand it, this was intended by the original author. I see the need to block only the first visit, but I also like the idea that a visit to any junk site should be as painful as possible for the entire duration. 

Of course, this can be subverted by SPA sites that never change the url, so insisting that this is a feature might not be the most consistent. Especially when it's actually a side effect of the implementation.

In short, I am not making any promises, but I _might_ implement an option to only block the first visit, so I'm leaving the issue open for now.

**Comment by keturn** [comment]
I'm finding that for infinite-scroll websites, this encourages very shallow interaction, as scrolling is free but clicking through to see the detailed view of a post or follow a thread is discouraged.

That's definitely a mixed bag.
